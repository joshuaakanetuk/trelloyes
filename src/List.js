import React from 'react';
import Card from './Card';
import './List.css';

class List extends React.Component {
    render() {
        const arrayOfCards = this.props.cards.map((elem, i) => {
            return <Card key={i} title={elem.title} content={elem.content} />
        });

        return (
            <section className="List">
                <header className="List-header">
                    <h2>{this.props.header}</h2>
                </header>
                <div className="List-cards">
                    {arrayOfCards}
                    <button type="button" className="List-add-button">
                        + Add Random Card
                </button>
                </div>
            </section>
        )
    }
}

export default List;