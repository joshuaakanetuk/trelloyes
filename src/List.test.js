import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header="List" cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' }]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
    .create(<List header="List"/>)
    .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});