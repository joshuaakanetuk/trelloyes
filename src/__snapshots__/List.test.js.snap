// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI as expected with no cards 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
